### ISSUE TITLE
Lambda layers

### ISSUE THREAD

**Issue by darasana910-cell** [sktime/sktime#8719]
Hello, the TapNetClassifier fails to load due to the use of Lambda layers. Could anyone provide guidance or suggestions on how to address this issue?

**Comment by fkiraly** [comment]
@darasana910-cell, can you provide a snippet of reproducible code for this?

**Comment by Ayaan-Skh** [comment]
Hii @fkiraly 
Can I work on this issue??

**Comment by NoorMajdoub** [comment]
hey @fkiraly this is caused by when you use TapNetClassifier's  build network function with use_rp=True , in this section of the code in sktime/networks/tapnet/_tapnet_tf.py line191
`
if self.use_rp:
                self.conv_1_models = keras.Sequential()

                for i in range(self.rp_group):
                    self.idx = np.random.permutation(input_shape[1])[0 : self.rp_dim]
                    channel = keras.layers.Lambda(
                        lambda x: tf.gather(x, indices=self.idx, axis=2)
                    )(input_layer)
`
The serialization of the inline lambda function during the loading /saving of the model breaks , it is better explained here https://keras.io/api/layers/core_layers/lambda/#:~:text=Lambda%20layers%20are%20saved%20by,overriding%20their%20get_config()%20method. 

it is triggered when you run sth like this 
`tapnet = TapNetClassifier(n_epochs=3, batch_size=4, use_rp=True, use_att=False,  verbose=False)
#use_rp true to trigger teh use of lambda layers
tapnet.fit(X_train, y_train)
with tempfile.TemporaryDirectory() as tmpdir:
    path = os.path.join(tmpdir, "model.keras")  
    tapnet.model_.save(path)                    #saving this model is what is gonna break it 
    loaded = keras.models.load_model(path) `
you get 

<img width="1245" height="223" alt="Image" src="https://github.com/user-attachments/assets/5f56e70c-cb8e-4e6e-8481-e15e6d13257b" />
A possible fix would be making a small class to do the same job as the lambda function and use it instead to avoid serialization  issues, i can make a pr with my fix , i tested it and it worked fine
